### DISCUSSION TITLE
API-Keys for user signins

### DISCUSSION THREAD

**Discussion by imchasingshadows** [ory/kratos#2759] [Q&A]
We are developing a frontend where users signin using Ory Kratos and have several microservices protected by Oathkeeper.

Some of these users will be implementing routes from Oathkeeper directly to speed up their use of our application by directly calling our services and skip the use of our frontend.

Is it possible to allow these users to create their own API key within kratos. So that they can later authenticate with kratos to get a valid token.

Or, in what other way could we implement such a flow.

**Comment by vinckr** [comment]
Hello @imchasingshadows 
Sorry for the late reply!
Have you found a solution since? If yes, would be interested how you solved it! 
Ory Kratos does not directly implement something like API keys, but I imagine you can make this kind of auth flow work using OAuth2. On Ory Network (the managed Ory service) you can generate API keys, but they are generally only used for admin tasks, not really as auth mechanism. 

Could the [client credentials OAuth2 flow](https://www.ory.sh/docs/oauth2-oidc/client-credentials) be a good solution for you?
